interview-mr/src/main/java/com/sovrn/interview/mr/AverageScore.java

```diff
@@ -32,26 +35,40 @@
  */
 public class AverageScore {
     public static void main(String[] args) throws Exception {
+        System.exit(runMapReduce(args[0], args[1]) ? 0 : 1);
+    }
+
+    // new entry point for testability
+    public static boolean runMapReduce(String input, String output) throws Exception {
         Configuration conf = new Configuration();
         Job job = Job.getInstance(conf, "Interview Averaging");
         job.setJarByClass(AverageScore.class);
         job.setMapperClass(AverageScoreMapper.class);
         job.setReducerClass(AverageScoreReducer.class);
-        // TODO: Finish the key and output setup
+        // Finish the key and output setup
+        job.setMapOutputKeyClass(Text.class);
+        job.setOutputKeyClass(Text.class);				// hostname
+        job.setOutputValueClass(FloatWritable.class);	// average score by hostname
+        
+        FileInputFormat.addInputPath(job, new Path(input));
+        FileOutputFormat.setOutputPath(job, new Path(output));
 
-        FileInputFormat.addInputPath(job, new Path(args[0]));
-        FileOutputFormat.setOutputPath(job, new Path(args[1]));
-
-        System.exit(job.waitForCompletion(true) ? 0 : 1);
+        return job.waitForCompletion(true);
     }
-
+    
     public static class AverageScoreMapper extends Mapper<Object, Text, Text, FloatWritable> {
 
         @Override
         protected void map(Object key, Text value, Mapper<Object, Text, Text, FloatWritable>.Context context)
                 throws IOException, InterruptedException {
-            // TODO : Implement the mapping phase
-            
+            // Implement the mapping phase
+            // parse input
+            WebDomainScore inputScore = WebDomainScore.fromTabSeparatedValues(value.toString());
+            // if we have both key and value
+            if (!Strings.isNullOrEmpty(inputScore.getHostname()) && inputScore.getScore() != null) {
```

Comment:
I skip any input if it's missing the hostname or the score. Seems better than inserting 0s which can skew averages a lot!